id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/util/CacheUtil.java

```diff
@@ -0,0 +1,45 @@
+package io.mosip.credential.request.generator.util;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import io.mosip.credential.request.generator.entity.CredentialEntity;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+public class CacheUtil {
+	
+    @Autowired
+    CacheManager cacheManager;
+    
+
+	  @Cacheable(value = "credential_transaction", key = "#requestId")
+	  public CredentialEntity setCredentialTransaction(String requestId, CredentialEntity credentialEntity) {
+	      return credentialEntity;
+	  }
+    
+	  @CacheEvict(value = "credential_transaction", key = "#requestId")
+	  private void evictCredentialTransaction(String requestId, CredentialEntity credentialEntity) {
+		  // TODO  This method doesn't need todo anything as of now. 
+	      return ;
+	  } 
+	  
+	  public void updateCredentialTransaction(String requestId, CredentialEntity credentialEntity) {
```

Comment:
why is the eviction required when the latest status is updated for the same key?

Reply:
I'm calling this method in update, cancel or retrigger credential methods, where status of credential is being updated anusha. so first I'm evicting the existing entry from cache and caching it with new status if entry already exists.